csrc/logical_domain_map.cpp

```diff
@@ -368,6 +368,52 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseLogicalDomainMap::map(
     return dom_map;
   }
 
+  // TODO: refactor to use getNonMappingDomainInfo instead.
+  if (GroupedMmaOp* op =
+          dynamic_cast<GroupedMmaOp*>(consumer_tv_->definition())) {
+    auto ndims_out = consumer_root.size();
+    if (producer_tv_->sameAs(op->matrix1())) {
+      // mapping m dimension;
```

Comment:
Shouldn't we also map k for mat1?

Reply:
sorry for the confusion caused by the other comment.

Reply:
It's not about comments. IIUC, in all the three cases, `k` appears in both `mat1` and `out` (as a reduction dimension) and therefore should be mapped. 

(Similar comments to go other branches)

Reply:
I need r{k} for TP. See insertreshardingspass where I use rfactor to decompose a row parallel linear to local and allreduce

Reply:
added r{k} at output.